From 35f4a49bfc47d15909e5906109cba0d8496be5c9 Mon Sep 17 00:00:00 2001
From: cooper_huang <cooper_huang@aspeedtech.com>
Date: Wed, 3 Jun 2026 14:29:44 +0800
Subject: [PATCH 2/2] mtd: spi-nor: Add XMC XM25QH01D flash part

Add support for the XMC XM25QH01D, a 1Gbit (128MiB) 3.3V SPI NOR flash.

  - JEDEC ID: 0x204021
  - Sector size: 64KiB with 4KiB subsector erase (SECT_4K)
  - Supports Dual and Quad read
  - Requires 4-byte address opcodes (SPI_NOR_4B_OPCODES)

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: cooper_huang <cooper_huang@aspeedtech.com>
Change-Id: I20fbbbe368c6647d059e9e23066e3133a5c0a380
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 drivers/mtd/spi/spi-nor-ids.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 5882eab2e3..1f68f6d203 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -361,6 +361,16 @@ const struct flash_info spi_nor_ids[] = {
 	/* XMC (Wuhan Xinxin Semiconductor Manufacturing Corp.) */
 	{ INFO("XM25QH64A", 0x207017, 0, 64 * 1024, 128, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("XM25QH128A", 0x207018, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ INFO("XM25QH256C", 0x204019, 0, 64 * 1024, 512, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+	{ INFO("XM25QU256C", 0x204119, 0, 64 * 1024, 512, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+	{ INFO("XM25QH512C", 0x204020, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+	{ INFO("XM25QU512C", 0x204120, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+	{ INFO("XM25QH01D", 0x204021, 0, 64 * 1024, 2048, SECT_4K | SPI_NOR_DUAL_READ |
+	       SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
 #endif
 	{ },
 };
-- 
2.43.0

